ivy/stateful/module.py

```diff
@@ -1141,48 +1153,468 @@ def build_mode(self):
         return self._build_mode
 
     @property
-    def built(self):
+    def built_(self):
         return self._built
 
-    # Methods #
-    # ------- #
+    # Module Converters #
+    # ------------------#
+    
+    @staticmethod
```

Comment:
I have implemented `to_haiku_model`as a static method to preserve the natural behavior of the user-written code.
If I chose the design format as used in (`to_torch_module()` and `to_keras_module()`), where the `ivy_module` is stored as an attribute of the class( i.e Composition in OOP terminology), the user would then have to explicitly pass the ivy_module instance when constructing the forward_fn() function.
Consider the following example:

ivy_model = IvyModel()
haiku_model = ivy_model.to_haiku_module()

**_Using static_method_**
def forward_fn(*a, **kw):
    model = haiku_model()
    return model(*a, **kw)

**_Using Composition(having ivy_model stored as an attribute)_**
def forward_fn(*a, **kw):
    model = haiku_model(ivy_model)  #will have to pass ivy_model explicitly
    return model(*a, **kw)

Is the current design implementation fine or should I go with the composition approach?

Reply:
I realise that this is now out-dated, but `to_haiku_model` should not be implemented as a static method, it should only be added as an instance method.